### PR TITLE
rename with arbitrary number of pairs

### LINKED ISSUE #129: add `rename(nda, :oldname=>:newname)`
It would be handy to be able to do renaming via giving the old and new name.
this would work like `Base.replace(::AbstractString, old=>new)`

Right now the only way to rename is to give a full list of what the new names should be

### PR DESCRIPTION
**Pull request by mzgubic**
Closes #129

### CODE REVIEW

File: src/name_core.jl [hunk 1]
```diff
@@ -127,6 +127,26 @@ function permute_dimnames(dimnames::NTuple{N, Symbol}, perm) where N
...
+"""
+function _rename(dimnames::Tuple, old_new::Vararg{Pair})
+    return ntuple(i -> _rename(dimnames[i], old_new...), length(dimnames))
+end
+
+function _rename(dimname::Symbol, old_new::Vararg{Pair})
+    # Avoid looping over pairs explicitly because that allocates.
+    nt = ntuple(i -> dimname === first(old_new[i]) ? i : 0, length(old_new))
+    which = sum(nt)
+    which > length(old_new) && throw(ArgumentError("`old_new` needs unique `old` items"))
```
Comment (oxinabox) [inline src/name_core.jl:146]: Can this error message be made clearer?
Maybe `duplicate old names are not permitted in rename` or something?>

**Comment by oxinabox** [comment]
CI is failing

**Comment by mzgubic** [comment]
> CI is failing

Yes, see: #128 and #112

File: src/name_core.jl (hunk 1 above)
Comment (mzgubic) [reply src/name_core.jl]: thanks, that's much better

**Comment by mzgubic** [comment]
@oxinabox CI passes now thanks for the `test_modern` macro

**Comment by oxinabox** [comment]
Tests on AV  fail because they are testing 1.1 which we don't actuall care about. 
See #133

**Review by oxinabox — APPROVED** [review]
Merge and tag when ready

**Pull request merged**